src/libraries/System.Text.Json/tests/Serialization/EnumConverterTests.cs

```diff
@@ -294,5 +295,222 @@ public enum MyEnum
             U = 1 << 20,
             V = 1 << 21,
         }
+
+        [Fact, OuterLoop]
+        public static void VeryLargeAmountOfEnumDictionaryKeysToSerialize()
+        {
+            // Ensure we don't throw OutOfMemoryException.
+
+            var options = new JsonSerializerOptions
+            {
+                Converters = { new JsonStringEnumConverter() }
+            };
```

Comment:
You don't have to use `JsonSerializerOptions` in this case. 

For `VeryLargeAmountOfEnumsToSerialize` the options are used because we don't serialize as string the enum values by default, you need to opt-in for it passing the `JsonStringEnumConverter`.

We do serialize as string by default the enum values when they are dictionary keys.

Reply:
Fixed 👍